### PR TITLE
[SYNPY-1420] Clean up otel spans

### PR DESCRIPTION
**Pull request by BryanFauble**
**Problem:**

1. OTEL span were a bit all over the place with varying depth and information

**Solution:**

1. Focus OTEL spans on the I/O portions of the application to know when requests are start/stopping or long running MD5 calculations are taking place

**Testing:**

1. Verified by running the upload benchmark script and checking a Jaeger container deployed on my local:
image

### CODE REVIEW

File: synapseclient/client.py
Comment (thomasyu888) [inline synapseclient/client.py:348]: Well, this is convenient.  Are we no longer using `requests`?

File: synapseclient/core/utils.py
Comment (thomasyu888) [inline synapseclient/core/utils.py:130]: Did you flip these intentionally?

File: synapseclient/client.py
Comment (BryanFauble) [reply synapseclient/client.py:348]: > Are we no longer using requests?

In all new code we're going to be using the httpx client. I cannot replace the requests library yet (For the previous code) because it would be a breaking change if someone were passing in their own client into the `Synapse()` class initializer.

> Basically, how I'm viewing this is that tracers are only provided by the httpx calls correct?

That's what I am moving towards - Yes. The caveat however is that I want a span for the overall Synapse entity doing actions. So a span is started for a Project, Folder, File, Table, Team ect.... that is being saved. Then inside of that span all network calls will have their span (So we can see start/end times) and I/O work like MD5 calculation will have a span.

image

File: synapseclient/core/utils.py
Comment (BryanFauble) [reply synapseclient/core/utils.py:130]: Yes, this is intentional. For MD5 calculation I only have 1 additional Python process that gets started up. I found some instability when I let the multi-processing library handle waiting on and scheduling all of these files - Which is why you see the `md5_semaphore` here. By placing the tracer within the `async with md5_semaphore` it means that the trace is only started when this file has gotten the semaphore. I am not capturing the time spent waiting to acquire the semaphore.

**Review by thomasyu888 — APPROVED** [review]
🔥 LGTM!  If someone else from @Sage-Bionetworks/dpe wants to take a look before merge, lets do so before EOD Friday :)

**Pull request merged**
